### PR TITLE
Update bfd for Poise 2.x

### PR DESCRIPTION
**Pull request by http-418**

### CODE REVIEW

**Comment by cbaenziger** [comment]
Changes look good! Thanks for the update. I'm just testing with `kitchen verify` now....

**Comment by http-418** [comment]
Now with passing tests and Ubuntu 16.04 support

File: libraries/bfd_beacon.rb [hunk 1]
```diff
@@ -41,7 +42,10 @@ def action_start
 
       converge_by("starting BFD #{new_resource.name}") do
         notifying_block do
-          bash "#{bfd_beacon}#{controls}#{listeners}"
+          code_snippet = "#{bfd_beacon}#{controls}#{listeners}"
```
Comment (cbaenziger) [inline libraries/bfd_beacon.rb:45]: Why the `code_snippet`; this looks to be used for the `name` and `code` but one could do:
`bash <command>`. (No need for the do and repetitive `code` attribute?)

File: recipes/default.rb [hunk 2]
```diff
@@ -43,27 +43,45 @@
...
-  notifies :restart, "service[bfdd-beacon]"
+if node[:lsb][:id] == 'Ubuntu' && node[:lsb][:release].to_f <= 14.04
+  template 'bfdd-beacon upstart config' do
+    path '/etc/init/bfdd-beacon.conf'
+    source 'bfdd-beacon.conf.erb'
+    owner 'root'
+    group 'root'
+  end
+else
+  template 'bfdd-beacon systemd config' do
```
Comment (cbaenziger) [inline recipes/default.rb:65]: Thanks for the SystemD support!

File: recipes/default.rb [hunk 3]
```diff
@@ -43,27 +43,45 @@
...
-  provider Chef::Provider::Service::Upstart
-  supports :status => true
+service 'bfdd-beacon' do
+  supports status: true
   action [:start]
+  if node[:lsb][:id] == 'Ubuntu' && node[:lsb][:release].to_f <= 14.04
+    provider Chef::Provider::Service::Upstart
+
+    if node[:lsb][:codename] == 'trusty'
+      ignore_failure true
```
Comment (cbaenziger) [inline recipes/default.rb]: Why is this needed? Removing it, test-kitchen still runs -- and we do want to know about errors with BFD!

File: libraries/bfd_beacon.rb (hunk 1 above)
Comment (http-418) [reply libraries/bfd_beacon.rb:45]: That is no longer supported as of recent versions of Chef 12.

File: recipes/default.rb (hunk 3 above)
Comment (http-418) [reply recipes/default.rb]: I can't re-create the error so I will remove this.
